taichi/rhi/llvm/llvm_device.h

```diff
@@ -20,6 +20,17 @@ class LlvmDevice : public Device {
     TI_NOT_IMPLEMENTED
   }
 
+  template <typename DEVICE, typename ALLOCINFO>
+  ALLOCINFO get_alloc_info(const DeviceAllocation handle) {
```

Comment:
I think providing a function `template <typename DEVICE> DEVICE *as()` and `device->as<CudaDevice>()->get_alloc_info` is better than this.